scripts/insert_snapshot_warning.js

```diff
@@ -0,0 +1,120 @@
+'use strict';
+
+const fs = require('fs');
+const { JSDOM } = require('jsdom');
+const { execSync } = require('child_process');
+
+const COMMIT = execSync('git rev-parse --verify HEAD').toString();
+
+const WARNING_HTML = `
+<div class="warning">
+  <details class="annoying-warning" open="">
+    <summary>This is a commit snapshot of the specification</summary>
+    <p>
+      This document contains the contents of the specification as of
+      <a href="https://github.com/tc39/ecma262/commit/${COMMIT}">commit ${COMMIT}</a>,
+      and should only be used as a historical reference. This commit may not
+      have even been merged into the specification.
+    </p>
+    <p>
+      Do not attempt to implement this version of the specification. Do not
+      reference this version as authoritative in any way. Instead, see
+      <a href="https://tc39.github.io/ecma262">https://tc39.github.io/ecma262</a>
+      for the living specification.
+    </p>
+  </details>
+</div>
+`;
+
+const WARNING_CSS = `
+<style>
+.warning::before { content: '\\26A0 Warning!'; }
+
+details.annoying-warning {
+  background-color: #920800;
+  background-image: linear-gradient(transparent 40%, rgba(255, 255, 255, 0.2));
+  border: solid rgba(0, 0, 0, 0.4);
+  border-radius: 3px;
+  border-width: 1px 1px 0 1px;
+  box-shadow: 0 0 0.5em rgba(0, 0, 0, 0.5);
+  color: rgba(255, 255, 255, 0.95);
+  opacity: .95;
+  position: fixed;
+  left: 5%;
+  margin: 0 auto;
+  right: 5%;
+  z-index: 10;
+}
+
+details.annoying-warning[open] {
+  top: 10%;
+  top: calc(5vw + 5vh);
+  max-width: 1024px;
+  outline: solid 10000px rgba(255, 255, 255, 0.6);
+}
+
+details.annoying-warning:not([open]) {
+  bottom: 0;
+  left: 0;
+  right: 0;
+  border-radius: 0;
+}
+
+details.annoying-warning > summary {
+  display: list-item; /* polyfill */
+  font-size: 0.875em;
+  font-weight: bold;
+  letter-spacing: 0.02em;
+  padding: 10px 5px;
+  text-align: center;
+  text-transform: uppercase;
+  text-shadow: 0px 1px 2px rgba(0, 0, 0, 0.85);
+  cursor: default;
+}
+
+details.annoying-warning > summary::after {
+  content: " Expand";
+  position: absolute;
+  top: 0;
+  right: 5px;
+  font-size: smaller;
+  font-weight: bold;
+}
+
+details.annoying-warning[open] > summary::after {
+  content: " Collapse";
+}
+
+details.annoying-warning p {
+  padding: 0 7.5% 1em;
+  line-height: 1.4;
+  margin: 0;
+  text-shadow: 0px 1px 1px rgba(0, 0, 0, 0.85);
+}
+
+details.annoying-warning a {
+  color: white;
+  text-decoration: underline;
+}
+</style>
+`;
+
+console.log('Inserting snapshot reference warning...');
+
+const dom = new JSDOM(fs.readFileSync('./out/index.html', 'utf8'));
```

Comment:
Don't convert to string; pass the buffer directly. See https://github.com/jsdom/jsdom#encoding-sniffing last paragraph